src/services/cache.ts

```diff
@@ -101,28 +101,29 @@ class CacheService {
         }
     }
 
-    private async tx(store: Store, write = false) {
-        return (await this.db()).transaction(store, write ? "readwrite" : "readonly");
+    private async tx<S extends Store, WriteMode extends boolean>(store: S, write: WriteMode = false as WriteMode): Promise<IDBPTransaction<unknown, [Store], WriteMode extends true ? "readwrite" : "readonly">> {
```

Comment:
Had to specify the return type to get the Mode parameter correct, so the put and delete methods were available when `true` was specified.